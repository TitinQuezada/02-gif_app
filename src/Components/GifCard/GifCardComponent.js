import React from 'react';
import './GifCard.Component.css';
import PropTypes from 'prop-types';

const GifCardComponent = ({ title, image }) => {
  return (
    <div className='card'>
      <img src={image} alt={title} height='100px'></img>
      <p>{title}</p>
    </div>
  );
};

GifCardComponent.propTypes = {
  title: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default GifCardComponent;
