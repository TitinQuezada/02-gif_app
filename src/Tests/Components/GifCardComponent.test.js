import React from 'react';
import { shallow } from 'enzyme';
import GifCardComponent from '../../Components/GifCard/GifCardComponent';
import '@testing-library/jest-dom';

describe('Gif Card Component Tests', () => {
  const gifTitle = 'titulo del gif';
  const gifUrl = 'http://imagenes.com/45';

  test('Check Gif Card Test Show Correctly', () => {
    //arrange
    const component = shallow(
      <GifCardComponent title={gifTitle} image={gifUrl} />
    );
    const { className } = component.find('div').props();
    const paragraphText = component.find('p').text();
    const { alt: imageAlt, src: srcImgae } = component.find('img').props();
    const classNames = className.split(' ');
    const cardClassContain = classNames.some(
      (className) => className === 'card'
    );

    //assert
    expect(component).toMatchSnapshot();
    expect(paragraphText).toBe(gifTitle);
    expect(imageAlt).toBe(gifTitle);
    expect(srcImgae).toBe(gifUrl);
    expect(cardClassContain).toBeTruthy();
  });
});
