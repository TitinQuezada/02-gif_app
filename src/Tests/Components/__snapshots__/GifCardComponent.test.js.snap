// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif Card Component Tests Check Gif Card Test Show Correctly 1`] = `
<div
  className="card"
>
  <img
    alt="titulo del gif"
    height="100px"
    src="http://imagenes.com/45"
  />
  <p>
    titulo del gif
  </p>
</div>
`;
